07.bowling_object/game.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  STRIKE = 'X'
+
+  def initialize(marks)
+    @marks = marks.split(',')
+  end
+
+  def score
+    frames = create_frames
+    frames.each_with_index.map do |frame, idx|
+      if idx < 9
+        frame.score + bonus(frames, frame, idx)
+      else
+        frame.score
+      end
+    end.sum
+  end
+
+  private
+
+  def create_frames
+    frames = Array.new(9) do
+      marks = @marks.first == STRIKE ? [@marks.shift] : @marks.shift(2)
+      Frame.new(*marks)
+    end
+    frames << Frame.new(*@marks)
+    frames
+  end
+
+  def bonus(frames, frame, idx)
+    next_frame = frames[idx + 1]
+    after_next_frame = frames[idx + 2]
+
+    if frame.strike? && next_frame.strike?
+      if idx < 8
+        10 + after_next_frame[0].point
+      else
+        10 + next_frame[1].point
+      end
```

Comment:
```suggestion
      bonus_shot = idx < 8 ? after_next_frame[0] : next_frame[1]
      10 + bonus_shot.point
```

みたいに書くと少しDRYになりそうです。